core/src/main/java/org/apache/iceberg/BaseFileScanTask.java

```diff
@@ -109,14 +110,11 @@ public String toString() {
     private final List<Long> offsets;
     private final List<Long> splitSizes;
     private final FileScanTask parentScanTask;
-    private final long targetSplitSize;
     private int sizeIdx = 0;
 
-    OffsetsAwareTargetSplitSizeScanTaskIterator(
-        List<Long> offsetList, FileScanTask parentScanTask, long targetSplitSize) {
+    OffsetsAwareTargetSplitSizeScanTaskIterator(List<Long> offsetList, FileScanTask parentScanTask) {
```

Comment:
Changing the behavior here to just always produce splits that are the size of a row group. Split-Size is not used here, only when recombining.